src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,138 @@
+package parser;
+
+import storage.Storage;
+import tasklist.TaskList;
+import ui.Ui;
+import mollyexception.MollyException;
+import task.*;
+public class Parser {
+
+    private Storage storage;
+
+    private TaskList taskList;
+
+    public Parser(Storage storage, TaskList taskList) {
+        this.storage = storage;
+        this.taskList = taskList;
+    }
+
+    /**
+     * Handles the different user inputs and decides what actions to perform for each of them.
+     * @param userInput
+     * @throws MollyException
+     */
+
+    public void handleUserInput(String userInput) throws MollyException {
```

Comment:
Any reason why you wrote a 100 line function instead of extracting out smaller functions?